wandb/sdk/data_types.py

```diff
@@ -1846,22 +1856,52 @@ def _set_initialization_meta(
                     boxes_final[key] = box_item
                 elif isinstance(box_item, dict):
                     boxes_final[key] = BoundingBoxes2D(box_item, key)
+
+                # move classes to image class object if any are missing from bbox
+                for key, name in boxes_final[key]._class_labels.items():
+                    classes_in_children = True
+                    if self._classes is not None and not self._classes.has_class(key):
+                        self._classes.set_class(key, name)
+
             self._boxes = boxes_final
 
         if masks:
             if not isinstance(masks, dict):
                 raise ValueError('Images "masks" argument must be a dictionary')
             masks_final: Dict[str, ImageMask] = {}
+
             for key in masks:
                 mask_item = masks[key]
                 if isinstance(mask_item, ImageMask):
                     masks_final[key] = mask_item
                 elif isinstance(mask_item, dict):
                     masks_final[key] = ImageMask(mask_item, key)
+
+                if "class_labels" not in masks_final[key]._val:
+                    continue
+
+                # move classes to image class object if any are missing from mask
+                for key, name in masks_final[key]._val["class_labels"].items():
+                    classes_in_children = True
+                    if self._classes is not None and not self._classes.has_class(key):
+                        self._classes.set_class(key, name)
+
             self._masks = masks_final
 
-        self._width, self._height = self.image.size  # type: ignore
-        self._free_ram()
+        if classes_in_children:
+            logging.warning(
+                "You have defined class sets in a box and/or mask. "
+                + "This method is deprecated and has been moved to an argument of the Image class."
```

Comment:
is it deprecated, or just handled differently?

Reply:
> discouragement of use of some terminology, feature, design, or practice, typically because it has been superseded or is no longer considered efficient or safe, without completely removing it or prohibiting its use https://en.wikipedia.org/wiki/Deprecation

We are discouraging the practice of setting classes at mask or box level. I think deprecate is an appropriate term, but I'm open to other suggestions